LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,256 @@
+﻿using Microsoft.Azure.Gaming.LocalMultiplayerAgent.Config;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using PlayFab;
+using PlayFab.MultiplayerModels;
+using System.IO;
+using Microsoft.Azure.Gaming.LocalMultiplayerAgent.DeploymentTool;
+using Microsoft.Azure.Storage.Blob;
+
+namespace Microsoft.Azure.Gaming.LocalMultiplayerAgent.MPSDeploymentTool
+{
+    public class DeploymentScript
+    {
+        private readonly MultiplayerSettings settings;
+        private readonly DeploymentSettings settingsDeployment;
+        public DeploymentScript(MultiplayerSettings multiplayerSettings)
+        {
+            settings = multiplayerSettings;
+            settingsDeployment = JsonConvert.DeserializeObject<DeploymentSettings>(File.ReadAllText("DeploymentTool/deployment.json"));
+        }
+
+        public async Task RunScriptAsync()
+        { 
+            PlayFabSettings.staticSettings.TitleId = settings.TitleId;
+
+            string secret = Environment.GetEnvironmentVariable("PF_SECRET");
+            if (string.IsNullOrEmpty(secret))
+            {
+                Console.WriteLine("Enter developer secret key");
+                PlayFabSettings.staticSettings.DeveloperSecretKey = Console.ReadLine();
+            }
+
+            var req = new PlayFab.AuthenticationModels.GetEntityTokenRequest();
+            var res = await PlayFabAuthenticationAPI.GetEntityTokenAsync(req);
+
+            if (res.Error != null)
+            {
+                Console.WriteLine(res.Error.ErrorMessage);
+                return;
+            }
+
+            // TODO: do validation checks
+
+            dynamic createBuild = null;
+            if (settings.RunContainer)
+            {
+                if (settingsDeployment.OSPlatform == "Windows")
+                {
+                    CreateBuildWithManagedContainerRequest request = GetManagedContainerRequest();
+
+                    foreach (var file in request.GameAssetReferences)
+                    {
+                        await CheckAssetFiles(file.FileName);
+                    }
+
+                    createBuild = await CreateBuildWithManagedContainer(request);
+                }
+                else
+                {
+                    //for CustomLinux Container
```

Comment:
if this needs to be filled in later mark it as a TODO, I think - or do a throw unsupportedexception or something like that for now. Can always add it later